src/reuse/modules/util/userSettings.ts

```diff
@@ -0,0 +1,129 @@
+"use strict";
+
+import { VerboseLoggerFactory } from "../../helper/verboseLogger";
+
+/**
+ * @class userSettings
+ * @memberof util
+ */
+
+export class UserSettings {
+  private vlf = new VerboseLoggerFactory("util", "user");
+  private _srvInstance = null;
+
+  private async initS4UserSettingService(user: string, password: string): Promise<any> {
+    if (!this._srvInstance) {
+      const vl = this.vlf.initLog(await this.initS4UserSettingService);
+      const params = browser.config.params;
+      if (params?.systemUrl) {
+        try {
+          this._srvInstance = await service.odata.init(`${params.systemUrl}/sap/opu/odata/UI2/INTEROP`, user, password);
+        } catch (error) {
+          if (error instanceof Error) {
+            throw new Error(`Failed to initialize S4 User Setting Service: ${error.message}.`);
+          } else {
+            throw new Error("Failed to initialize S4 User Setting Service.");
+          }
+        }
+      } else {
+        throw new Error("System URL is missing in the config file.");
+      }
+    }
+  }
+
+  public async setLanguageFromUserSettings(user: string, password: string) {
+    const vl = this.vlf.initLog(this.setLanguageFromUserSettings);
+    await this.initS4UserSettingService(user, password);
+    const res = await service.odata.get(this._srvInstance, "UserProfileProperties", { id: "LANGUAGE", shellType: "FLP" });
```

Comment:
I would try catch all the API calls as you did for the init.

Reply:
move all api calls in _getXXXXresponse functions and added try catch. Error handling will be improved later on in different PR as we agreed